internal/wasm/sys_test.go

```diff
@@ -147,3 +150,57 @@ func TestNewSysContext_Environ(t *testing.T) {
 		})
 	}
 }
+
+func TestSysContext_Close(t *testing.T) {
+	t.Run("no files", func(t *testing.T) {
+		sys := DefaultSysContext()
+		require.NoError(t, sys.Close())
+	})
+
+	t.Run("open files", func(t *testing.T) {
+		tempDir := t.TempDir()
+		pathName := "test"
+		file, _ := createWriteableFile(t, tempDir, pathName, make([]byte, 0))
+
+		sys, err := NewSysContext(
+			0,   // max
+			nil, // args
+			nil, // environ
+			nil, // stdin
+			nil, // stdout
+			nil, // stderr
+			map[uint32]*FileEntry{ // openedFiles
+				3: {Path: "."},
+				4: {Path: path.Join(".", pathName), File: file},
+			},
+		)
+		require.NoError(t, err)
+
+		// Closing should delete the file descriptors after closing the files.
+		require.NoError(t, sys.Close())
+		require.Empty(t, sys.openedFiles)
+
+		// Verify it was actually closed, by trying to close it again.
+		err = file.Close()
```

Comment:
another side-effect you can use. Ex. if you can't easily reach the store or the runtime.. if you can open a file, you can check that it closed, by trying to close it again!